app/process/handler/read-only/nft/fee/handler.go

```diff
@@ -74,9 +77,15 @@ func NewHandler(
 		readOnlyService:    readOnlyService,
 		hederaNftFees:      hederaNftFees,
 	}
+
+	event.On(constants.EventBridgeConfigUpdate, event.ListenerFunc(func(e event.Event) error {
+		return bridgeCfgEventHandler(e, instance)
+	}), constants.HandlerEventPriority)
+
+	return instance
```

Comment:
This handler should not use `hederaNftFees` at all.
Line `validFee := fmh.distributor.ValidAmount(fmh.hederaNftFees[transferMsg.SourceAsset])`
Should be resolved to `validFee := fmh.distributor.ValidAmount(transferMsg.Fee)`

Reply:
This way you will not need to have an event listener.